### PR TITLE
Exit from catalogue engage if commit cannot be made

### LINKED ISSUE #110: git commit by catalogue engage gives error if no previous commits have been made
Something of a niche bug - I came across this one in a quick test this afternoon.

I created a new git repository with `git init`, saved a file in the code folder and then tried to run `catalogue engage ...`. The option to commit all uncommitted changes was presented, and after selecting `y`, the following error appeared:
```
gitdb.exc.BadName: Ref 'HEAD' did not resolve to an object
```

I'd assume this is because repro-catalogue tries to create a new branch for this commit, but as there are no prior commits it isn't possible to create a branch yet. We should check for this and either
1. Give the option to commit directly to master
2. Exit and make the user manually handle the first commit

Option 2. requires a little more effort on the user's part, but would mean that repro-catalogue won't change the master branch which I think is what we want here.

### PR DESCRIPTION
**Pull request by LouiseABowler**
Fixes #110: If user tries to run `catalogue engage` when the code folder is a git repository but no commits have been made yet, they are given the option for catalogue to commit their current files. However, the lack of an initial commit means that an error is generated, with little helpful information if the person is new to git.

The outputs from catalogue engage in this workflow will now be as follows:
```
Working directory contains uncommitted changes.
Do you want to stage and commit all changes? (y/[n])
y

Cannot create a branch for this commit as there are no existing commits.
Make a commit manually, then run catalogue engage again.
```

### CODE REVIEW

**Comment by edaub** [comment]
Louise, is this done and ready for review? I may have missed it and want to check if you are waiting on a review from me (or someone else).

**Comment by LouiseABowler** [comment]
Yes, this one's done @edaub - would be great if you could check if you have time, but it's not the most urgent one 🙂

File: tests/test_engage.py
Comment (edaub) [inline tests/test_engage.py]: Minor stylistic point: `assert not git_query(git_repo_no_commits, "catalogue_results")`

File: tests/test_engage.py
Comment (edaub) [inline tests/test_engage.py]: Same as above: `assert not git_query(git_repo_no_commits, "catalogue_results", True)`

File: tests/test_engage.py
Comment (edaub) [inline tests/test_engage.py]: Same as above.

File: catalogue/engage.py
Comment (edaub) [inline catalogue/engage.py]: You don't actually use `e` anywhere here -- did you mean to use it in the error message? I think it's fine the way it is, so I think you can just do `except BadName:`

**Review by edaub — APPROVED** [review]
A few minor coding style points, but I think this is great and I'm glad you caught this (I certainly didn't think of it before!).

**Comment by kevinxufs** [comment]
Hi @edaub @LouiseABowler 

I think we mentioned that we would look to work on the JOSS article after these three PRs have been merged.

I've just merged the two config PRs, so we're just waiting on this one.

Unfortunately, it looks like the config PR has led to a conflict here!

**Comment by edaub** [comment]
@kevinxufs Merge conflicts like this when you have a few issues open simultaneously is totally normal -- usually in this case it would be up to @LouiseABowler to fix the conflicts when merging, though anyone in principle should be able to fix as long as they have write access to the repo.

EDIT: Looking at the conflict, the diff isn't clean at all, so this is probably slightly more complicated, though my guess is that we just have a couple of things to resolve. I don't think this will hold up the JOSS submission at all!

**Comment by kevinxufs** [comment]
> 
> 
> @kevinxufs Merge conflicts like this when you have a few issues open simultaneously is totally normal -- usually in this case it would be up to @LouiseABowler to fix the conflicts when merging, though anyone in principle should be able to fix as long as they have write access to the repo.
> 
> EDIT: Looking at the conflict, the diff isn't clean at all, so this is probably slightly more complicated, though my guess is that we just have a couple of things to resolve. I don't think this will hold up the JOSS submission at all!

Okay - hopefully Louise can figure this out.

I have a bit of time today so I can try and get a few things rolling for the submission before I leave.

**Comment by edaub** [comment]
Fixed conflicts and merging.

**Pull request merged**
